src/compass/s1_geo_stack.py

```diff
@@ -0,0 +1,566 @@
+
+        # List all zip file and extract info
+        zip_file_list = sorted(glob.glob(f'{slc_dir}/S1*zip'))
+
+        for zip_file in zip_file_list:
+            sensor_id, _, start_datetime, \
+            end_datetime, _ = parse_safe_filename(zip_file)
+
+            # Find precise orbits first
+            orbit_dict = get_orbit_dict(sensor_id, start_datetime,
+                                        end_datetime, 'AUX_POEORB')
+            # If orbit_dict is empty, precise orbits have not been found. Find restituted orbits instead
+            if orbit_dict == None:
+                orbit_dict = get_orbit_dict(sensor_id, start_datetime,
+                                            end_datetime, 'AUX_RESORB')
+            # Download the orbit file
+            download_orbit(orbit_dir, orbit_dict['orbit_url'])
+
+    # Generate burst map and prune it if a list of burst ID is provided
+    zip_file = sorted(glob.glob(f'{slc_dir}/S1*zip'))
+    burst_map = generate_burst_map(zip_file, orbit_dir, x_spac, y_spac, epsg)
+
+    # Identify burst IDs common across the stack and remove from the dataframe
+    # burst IDs that are not in common
+    common_ids = get_common_burst_ids(burst_map)
+    burst_map = prune_dataframe(burst_map, 'burst_id', common_ids)
+
+    # If user selects burst IDs to process, prune unnecessary bursts
+    if burst_id is not None:
+        burst_map = prune_dataframe(burst_map, 'burst_id', burst_id)
+
+    # Select only dates between start and end
+    if start_date is not None:
+        burst_map = burst_map[burst_map['date'] >= start_date]
+    if end_date is not None:
+        burst_map = burst_map[burst_map['date'] <= end_date]
+
+    # Exclude some dates if the user requires it
+    if exclude_dates is not None:
+        burst_map = prune_dataframe(burst_map, 'date', exclude_dates)
+
+    # Ready to geocode bursts
+    for safe in zip_file:
+        i_subswath = [1, 2, 3]
+        orbit_path = get_orbit_file_from_list(safe,
+                                              glob.glob(f'{orbit_dir}/S1*'))
+        for subswath in i_subswath:
+            bursts = load_bursts(safe, orbit_path, subswath)
+            for burst in bursts:
+                date = int(burst.sensing_start.strftime("%Y%m%d"))
+                if (burst.burst_id in list(set(burst_map['burst_id']))) and \
+                        (date in list(set(burst_map['date']))):
+                    runconfig_path = create_runconfig(burst, safe, orbit_path,
+                                                      dem_file,
+                                                      work_dir, burst_map,
+                                                      flatten,
+                                                      is_split_spectrum,
+                                                      low_band, high_band, pol,
+                                                      x_spac, y_spac)
+                    date_str = str(date)
```

Comment:
I think this line should be moved to right after line 534. Right now if the "else" triggers, the logging entry on line 552 uses `date_str`, but it isn't defined.